### ISSUE TITLE
JEP 447  improve error message for constructor calls occurring nested in some control structure

### ISSUE THREAD

**Issue by stephan-herrmann** [eclipse-jdt/eclipse.jdt.core#2468]
```
public class TestFlow {
	TestFlow(boolean f) {
		if (f)
			super();
		else
			this();
	}
}
```
This reports "Constructor call must be the first statement in a constructor", despite the preview being enabled. 

In JLS this is managed by grammar rules admitting constructor calls only as a top level statement in the constructor body. When this feature enabled, should we say smth like "must be a top-level statement in a constructor body"? Would that be clear, or do we need to explicitly mention what is not allowed: wrapping the constructor call in some control structure or block?

**Comment by stephan-herrmann** [comment]
Found this comment in a recent spec draft:
> The grammar makes it impossible, for example, to place explicit constructor invocations in different branches of an if statement.

**Comment by stephan-herrmann** [comment]
> should we say smth like "must be a top-level statement in a constructor body"? Would that be clear, or do we need to explicitly mention what is not allowed: wrapping the constructor call in some control structure or block?

In #2472 I opted for a simply "Constructor call is not allowed here" (javac says "Explicit onstructor call is not allowed here")

**Comment by stephan-herrmann** [comment]
Resolved via #2685
